packages/outbox-core/lib/outbox.spec.ts

```diff
@@ -0,0 +1,327 @@
+import { randomUUID } from 'node:crypto'
+import {
+  CommonMetadataFiller,
+  DomainEventEmitter,
+  EventRegistry,
+} from '@message-queue-toolkit/core'
+import {
+  type CommonEventDefinition,
+  type CommonEventDefinitionPublisherSchemaType,
+  enrichMessageSchemaWithBase,
+} from '@message-queue-toolkit/schemas'
+import pino, { type Logger } from 'pino'
+import { afterEach, beforeEach, describe, expect, it, vi } from 'vitest'
+import { z } from 'zod'
+import { InMemoryOutboxAccumulator, type OutboxAccumulator } from './accumulators'
+import type { OutboxEntry } from './objects'
+import { type OutboxDependencies, OutboxEventEmitter, OutboxProcessor } from './outbox'
+import type { OutboxStorage } from './storage'
+
+const TestEvents = {
+  created: {
+    ...enrichMessageSchemaWithBase(
+      'entity.created',
+      z.object({
+        message: z.string(),
+      }),
+    ),
+  },
+
+  updated: {
+    ...enrichMessageSchemaWithBase(
+      'entity.updated',
+      z.object({
+        message: z.string(),
+      }),
+    ),
+  },
+} as const satisfies Record<string, CommonEventDefinition>
+
+type TestEventsType = (typeof TestEvents)[keyof typeof TestEvents][]
+
+const createdEventPayload: CommonEventDefinitionPublisherSchemaType<typeof TestEvents.created> = {
+  payload: {
+    message: 'msg',
+  },
+  type: 'entity.created',
+  metadata: {
+    originatedFrom: 'service',
+    producedBy: 'producer',
+    schemaVersion: '1',
+    correlationId: randomUUID(),
+  },
+}
+
+const TestLogger: Logger = pino()
+
+class InMemoryOutboxStorage<SupportedEvents extends CommonEventDefinition[]>
```

Comment:
feels a bit cluttered with this inlined within the test, can we move it to a separate test directory in its own file?

Reply:
Moved test + impl to `test/`